core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -399,6 +421,19 @@ private String loadTranslogUUIDFromCommit(IndexWriter writer) throws IOException
         }
     }
 
+    /**
+     * Reads the current stored history ID from the IW commit data. If the id is not found, returns null.
+     */
+    @Nullable
+    private String loadHistoryUUIDFromCommit(final IndexWriter writer) throws IOException {
+        String uuid = commitDataAsMap(writer).get(HISTORY_UUID_KEY);
+        if (uuid == null) {
+            assert config().getIndexSettings().getIndexVersionCreated().before(Version.V_6_0_0_rc1) :
```

Comment:
should we make this a hard fail? I wonder if we should return a null uuid here it will just fail later?!

Reply:
yeah, can do - it shouldn't happen anyway. I tend to communicate this "if this happens it's a bug" with assertions but I'm happy to convert to an IllegalState or something.

Reply:
yeah ++

Reply:
On a second thought - assertions cause nodes to die, which means our integration tests will be stopped on the spot - with an illegal state, the nodes we recover by doing another recovery and we'll never see the failure.

Reply:
fair enough